### ISSUE TITLE
Bullet sometimes gets stuck and won't fire.

### ISSUE THREAD

**Issue by stamarty** [stamarty/Space-Invaders#1]
Sometimes when pushing space-bar to fire the bullet the bullet will get stuck. The noise happens, but the bullet won't actually move.

**Comment by RookTKO** [comment]
It's an issue with how the fire_bullet function gets handled.

**Comment by stamarty** [comment]
Yeah. I think its something to do with how it changes state. I haven't gotten back in to fiddle with it in a few days.

**Comment by RookTKO** [comment]
At some point it gets undefined. It's actually super frustrating, I can't seem to find where it gets lost and gets set to undefined.
